data/lua/core/unit_test.lua

```diff
@@ -0,0 +1,166 @@
+
+--[========[Unit Test Helpers]========]
+
+if rawget(_G, 'unit_test') ~= nil then
+	print("Loading unit_test module...")
+	
+	--! Causes the test to complete with a specific result.
+	--! This is mostly an internal detail but might very occasionally be useful to produce a result other than pass or fail
+	function unit_test.finish(result)
+		wesnoth.wml_actions.endlevel{
+			test_result = result,
+			linger_mode = true
+		}
+	end
+
+	--! End the test in unconditional failure
+	function unit_test.fail()
+		unit_test.finish('fail')
+	end
+
+	--! End the test in success if all assertions passed, otherwise in failure
```

Comment:
Please document somewhere that the reason for needing this is so that a test can be spread over multiple events (even over multiple turns).